process-engine/process_engine_core
# 🔒 Prevent use of misconfigured events

## Changes

1. Reduce code duplication in EventParser
2. Only allow one EventDefinition for each event
3. Add sanity checks for MessageEvents, SignalEvents and TimerEvents
4. Restructure EventDefinition assignment on EventParser to prevent unnecessary parsing operations
5. Add UnitTests for EventParser

## Issues
 
Part of https://github.com/process-engine/process_engine_runtime/issues/437

PR: #303

## How to test the changes

- Try deploying ProcessModels with MessageEvents or SignalEvents that are missing references to a message or signal
- Try deploying ProcessModels with TimerEvents that are missing a timer type or value

Each of these cases should fail. Everything else should work.
